isovar/locus_reads.py

```diff
@@ -29,17 +28,34 @@
 )
 from .common import list_to_string
 from .dataframe_builder import DataFrameBuilder
-
+from .value_object import ValueObject
 
 logger = logging.getLogger(__name__)
 
-class LocusRead(namedtuple("LocusRead", [
+class LocusRead(ValueObject):
+    __slots__ = [
         "name",
         "sequence",
         "reference_positions",
         "quality_scores",
         "base0_read_position_before_variant",
-        "base0_read_position_after_variant"])):
+        "base0_read_position_after_variant"
+    ]
+
+    def __init__(
```

Comment:
@ihodes I guess the counter-argument is that in most Python code you'd expect to see an `__init__` method where all the fields get set explicitly, thus it's surprising to have it happen automatically. 

Also, much more broadly, how do you, @julia326, @tavinathanson, @armish, and @hammer feel about using the relevant parts of scikit-learn's [coding guidelines](http://scikit-learn.org/stable/developers/contributing.html#coding-guidelines) for our Python projects?

With regard to initialization methods they say:

>In addition, every keyword argument accepted by __init__ should correspond to an attribute on the instance. 
...
>There should be no logic, not even input validation, and the parameters should not be changed. The corresponding logic should be put where the parameters are used, typically in `fit`.